src/var.c

```diff
@@ -53,8 +56,86 @@ struct fi_var_t {
 
 static struct fi_var_t *fi_vars = NULL;
 
+static int fi_var_get(const struct fi_provider *provider, const char *var_name,
+		char **value)
+{
+	struct fi_var_t *v;
+
+	// Check for bozo cases
+	if (var_name == NULL || value == NULL) {
+		FI_DBG(provider, FI_LOG_CORE,
+			"Failed to read %s variable: provider coding error\n",
+			var_name);
+		return -FI_EINVAL;
+	}
+
+	for (v = fi_vars; v; v = v->next) {
+		if (strcmp(v->provider->name, provider->name) == 0 &&
+			strcmp(v->var_name, var_name) == 0) {
+			*value = getenv(v->env_var_name);
+
+			return FI_SUCCESS;
+		}
+	}
+
+	FI_DBG(provider, FI_LOG_CORE,
+		"Failed to read %s variable: was not registered\n", var_name);
+	return -FI_ENOENT;
+}
+
+__attribute__((visibility ("default")))
+int DEFAULT_SYMVER_PRE(fi_getsettings)(struct fi_setting **vars, int *count) {
+
+	struct fi_setting *vhead = NULL;
+	struct fi_var_t *ptr;
+	int ret, len = 0, i = 0;
+	char *tmp;
+
+	// just get a count
+	for (ptr = fi_vars; ptr; ptr = ptr->next, ++len);
```

Comment:
Might have gotten lost because I commented right before you re-pushed, but this format is nice IMHO:

```c
    for (ptr = fi_vars; ptr; ptr = ptr->next, ++len)
        continue;
```

You still see an obvious block under the for line, etc.

But that's just my $0.02.  :smile: